From 38581b29a95f542e5696f436fea874a881f75825 Mon Sep 17 00:00:00 2001
From: Cesar Eduardo Barros <cesarb@cesarb.net>
Date: Mon, 4 Aug 2008 08:34:50 +0100
Subject: [PATCH] pcf50606: fix gllin compat link

Uses bus_create_device_link to correctly create the gllin compat link.

Signed-off-by: Cesar Eduardo Barros <cesarb@cesarb.net>
---
 drivers/i2c/chips/pcf50606.c |   17 +++--------------
 1 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/i2c/chips/pcf50606.c b/drivers/i2c/chips/pcf50606.c
index 4218917..c97cad3 100644
--- a/drivers/i2c/chips/pcf50606.c
+++ b/drivers/i2c/chips/pcf50606.c
@@ -1784,12 +1784,6 @@ static int pcf50606_detect(struct i2c_adapter *adapter, int address, int kind)
 
 #ifdef CONFIG_MACH_NEO1973_GTA01
 	if (machine_is_neo1973_gta01()) {
-		/* FIXME: what is the device name we are symlinking to?
-		 * symlink won't get created until this is fixed.
-		 */
-		struct device * busdev = bus_find_device_by_name(
-							&platform_bus_type,
-							NULL, "FIXME");
 		gta01_pm_gps_dev.dev.parent = &new_client->dev;
 		switch (system_rev) {
 		case GTA01Bv2_SYSTEM_REV:
@@ -1801,16 +1795,11 @@ static int pcf50606_detect(struct i2c_adapter *adapter, int address, int kind)
 		}
 		platform_device_register(&gta01_pm_gps_dev);
 		/* a link for gllin compatibility */
-		/* 2.6.25 tracking: platform_bus_type.devices that was here
-		 * has gone away in mainline
-		 */
-		if (busdev) {
-			err = sysfs_create_link(&busdev->kobj,
+		err = bus_create_device_link(&platform_bus_type,
 			&gta01_pm_gps_dev.dev.kobj, "gta01-pm-gps.0");
-			if (err)
-				printk(KERN_ERR
+		if (err)
+			printk(KERN_ERR
 			    "sysfs_create_link (gta01-pm-gps.0): %d\n", err);
-		}
 	}
 #endif
 
-- 
1.5.6.3

